examples/require.ensure-handling-error/README.md

```diff
@@ -0,0 +1,304 @@
+ * @author "Evgeny Reznichenko" <kusakyky@gmail.com>
+ */
+
+
+require.ensure(["./a.js"], function (error, require) {
+    if (error) {
+        // do something on error
+        console.error(error);
+    } else {
+        require("./a.js");
+    }
+}, "a");
+```
+
+# a.js
+
+``` javascript
+/**
+ * MIT License http://www.opensource.org/licenses/mit-license.php
+ * @author "Evgeny Reznichenko" <kusakyky@gmail.com>
+ */
+
+console.log("a");
+```
+
+# webpack.config.js
+
+``` javascript
+var path = require('path');
+var JsonpErrorHandlingPlugin = require("../../lib/JsonpErrorHandlingPlugin");
+
+module.exports = {
+	entry: "./example",
+	output: {
+		path: path.join(__dirname, "js"),
+		filename: "output.js",
+		chunkFilename: "[name].chunk.output.js",
+		jsonpLoadTimeout: 10 * 1000 // default 60 sec
+	},
+	plugins: [
+		new JsonpErrorHandlingPlugin()
+	]
+}
+```
+
+# js/output.js
+
+``` javascript
+/******/ (function(modules) { // webpackBootstrap
+/******/ 	// install a JSONP callback for chunk loading
+/******/ 	var parentJsonpFunction = window["webpackJsonp"];
+/******/ 	window["webpackJsonp"] = function webpackJsonpCallback(chunkIds, moreModules) {
+/******/ 		// add "moreModules" to the modules object,
+/******/ 		// then flag all "chunkIds" as loaded and fire callback
+/******/ 		var moduleId, chunkId, i = 0, callbacks = [];
+/******/ 		for(;i < chunkIds.length; i++) {
+/******/ 			chunkId = chunkIds[i];
+/******/ 			if(installedChunks[chunkId])
+/******/ 				callbacks.push.apply(callbacks, installedChunks[chunkId]);
+/******/ 			installedChunks[chunkId] = 0;
+/******/ 		}
+/******/ 		for(moduleId in moreModules) {
+/******/ 			modules[moduleId] = moreModules[moduleId];
+/******/ 		}
+/******/ 		if(parentJsonpFunction) parentJsonpFunction(chunkIds, moreModules);
+/******/ 		while(callbacks.length)
+/******/ 			callbacks.shift().call(null, null, __webpack_require__);
+/******/
+/******/ 	};
+/******/
+/******/ 	// setupScriptLoadErrorHandler
+/******/ 	function setupScriptLoadErrorHandler(script, chunkId) {
+/******/ 		var timeoutId;
+/******/ 		var done = false;
+/******/
+/******/ 		function end(error) {
+/******/ 			script.onerror = script.onload = script.onreadystatechange = null;
+/******/ 			clearTimeout(timeoutId);
+/******/ 			if (done) return;
+/******/ 			if (error) {
+/******/ 				installedChunks[chunkId] = -1;
+/******/ 				chunkLoadErrors[chunkId] = error;
+/******/ 				var callbacks = installedChunks[chunkId];
+/******/ 				delete installedChunks[chunkId];
+/******/
+/******/ 				if (callbacks) while(callbacks.length) {
+/******/ 					callbacks.shift().call(null, chunkLoadErrors[chunkId], __webpack_require__);
+/******/ 				}
+/******/ 			}
+/******/ 			done = true;
+/******/ 		}
+/******/
+/******/ 		script.onload = script.onreadystatechange = function() {
+/******/ 			var readyState = this.readyState;
+/******/ 			if (!readyState || readyState === "loaded" || readyState === "complete") {
+/******/ 				end();
```

Comment:
Looking at the [requirejs docs](http://requirejs.org/docs/api.html#errors) they reckon older IE's will call the `onreadystatechange` handler with `this.readyState` set to 'complete' even if the script 404'd. I haven't got access to IE to try this out at the moment but I think it might be worth testing this and as a potential fix check for `installedChunks[chunkId] === 0` as you can be sure the script has downloaded successfully if the jsonp callback has been run.